src/SAWScript/Crucible/MIR/ResolveSetupValue.hs

```diff
@@ -334,6 +361,28 @@ resolveSetupVal mcc env tyenv nameEnv val =
       let (fldShpAssn, valAssn) = Ctx.unzip fldAssn
       let tupleShp = TupleShape (Mir.TyTuple fldMirTys) fldMirTys fldShpAssn
       pure $ MIRVal tupleShp valAssn
+    MS.SetupSlice slice ->
+      case slice of
+        MirSetupSliceRaw{} ->
+          panic "resolveSetupVal" ["MirSetupSliceRaw not yet implemented"]
+        MirSetupSlice arrRef -> do
+          SetupSliceFromArray _elemTpr sliceShp refVal len <-
+            resolveSetupSliceFromArray bak arrRef
+          lenVal <- usizeBvLit sym len
+          pure $ MIRVal sliceShp (Ctx.Empty Ctx.:> RV refVal Ctx.:> RV lenVal)
+        MirSetupSliceRange arrRef start end -> do
+          unless (start <= end) $
+            fail $ "slice index starts at " ++ show start
+                ++ " but ends at " ++ show end
+          SetupSliceFromArray elemTpr sliceShp refVal0 len <-
+            resolveSetupSliceFromArray bak arrRef
+          unless (end < len) $
+            fail $ "range end index " ++ show end
+                ++ " out of range for slice of length " ++ show len
```

Comment:
```suggestion
          unless (end <= len) $
            fail $ "range end index " ++ show end
                ++ " out of range for slice of length " ++ show len
```

Reply:
Oops, I'm not sure how I managed to convince myself that it should be `end < len` instead of `end <= len`. Fixed.